extensions/map-layers/src/ui/widget/MapUrlDialog.tsx

```diff
@@ -100,16 +137,55 @@ export function MapUrlDialog({ isOverlay, onOkResult, mapTypesOptions }: { isOve
   }, [setMapUrl]);
 
   const handleOk = React.useCallback(() => {
-    if (mapUrl && mapName) {
+    if (layerToEdit && layerIdxToEdit !== undefined && props.layerToEdit && props.activeViewport?.displayStyle) {
       const source = MapLayerSource.fromJSON({
         url: mapUrl,
         name: mapName,
         formatId: mapType,
         userName: wmsServerUsername,
         password: wmsServerPassword,
       });
-      if (source)
-        doAttach(source);
+
+      if (source) {
+
+        source.validateSource().then(async (validation) => {
+          if (validation.status === MapLayerSourceStatus.Valid && props?.activeViewport) {
+            source.subLayers = validation.subLayers;
+
+            props.activeViewport.displayStyle.changeMapLayerProps({
```

Comment:
After user provided credentials by editing the layer definition, we inject the credentials in the display style and restore the layer status to Valid.  Layer sill starting fetching tiles again.